vllm/engine/protocol.py

```diff
@@ -69,24 +70,50 @@ async def beam_search(
         ignore_eos = params.ignore_eos
         temperature = params.temperature
         length_penalty = params.length_penalty
+        include_stop_str_in_output = params.include_stop_str_in_output
 
         tokenizer = await self.get_tokenizer(lora_request=None)
-        tokenizedPrompt = prompt if isinstance(
-            prompt, list) else tokenizer.encode(prompt)
-        tokenizedLength = len(tokenizedPrompt)
+
+        if isinstance(prompt, dict):
+            if "prompt" in prompt:
+                tokenized_prompt = tokenizer.encode(prompt.get("prompt"))
+                multi_modal_data = prompt.get("multi_modal_data")
+                mm_processor_kwargs = prompt.get("mm_processor_kwargs")
+            elif "prompt_token_ids" in prompt:
+                tokenized_prompt = prompt.get("prompt_token_ids")
+                multi_modal_data = prompt.get("multi_modal_data")
+                mm_processor_kwargs = prompt.get("mm_processor_kwargs")
+            else:
+                raise TypeError(
+                    "Dictionary input must be a TextPrompt or TokensPrompt")
+        else:
+            tokenized_prompt = prompt if isinstance(
+                prompt, list) else tokenizer.encode(prompt)
+            multi_modal_data = None
+            mm_processor_kwargs = None
```

Comment:
We can use `InputPreprocessor._extract_prompt_components` here.

Reply:
Thank you! I changed to this function